bin/strata-cli/src/cmd/faucet.rs

```diff
@@ -70,13 +70,76 @@ impl fmt::Display for Chain {
 }
 
 pub async fn faucet(args: FaucetArgs, seed: Seed, settings: Settings) {
+    #[cfg(feature = "strata_faucet")]
+    let network_type = net_type_or_exit(&args.network_type);
+
+    #[cfg(feature = "strata_faucet")]
+    let (address, claim) = match network_type {
+        NetworkType::Signet => {
+            let mut l1w =
+                SignetWallet::new(&seed, settings.network, settings.signet_backend.clone())
+                    .unwrap();
+            let address = match args.address {
+                None => {
+                    let address_info = l1w.reveal_next_address(KeychainKind::External);
+                    l1w.persist().unwrap();
+                    address_info.address
+                }
+                Some(address) => {
+                    let address = Address::from_str(&address).unwrap_or_else(|_| {
+                        eprintln!("Invalid signet address provided as argument.");
+                        std::process::exit(1);
+                    });
+                    address
+                        .require_network(settings.network)
+                        .expect("wrong bitcoin network")
+                }
+            };
+            (address.to_string(), "claim_l1")
+        }
+        NetworkType::Strata => {
+            let l2w = StrataWallet::new(&seed, &settings.strata_endpoint).unwrap();
+            // they said EVMs were advanced 👁️👁️
```

Comment:
This comment is not needed anymore it was a hot take on alloy's huge trait bounds on generics.